hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/SingleTransactionStreamRecord.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.state;
+
+import com.hedera.hapi.block.stream.BlockItem;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.stream.Stream;
+
+/**
+ * This class is a record of a stream of single transaction records and a stream of block items.
+ * It is used to store the streams of single transaction records and block items in a single object.
+ */
+public record SingleTransactionStreamRecord(
+        @Nullable Stream<SingleTransactionRecord> singleTransactionRecordStream,
+        @Nullable Stream<BlockItem> blockItemStream) {}
```

Comment:
This is the same model that `SingleTransactionRecordBuilderImpl` follows, specifically that there's no indication of which properties are set until runtime (`SingleTr...` simply implements a superset of all the properties). While it's generally good to be consistent, I've found reasoning about the transaction builder to be very difficult. I think we would benefit significantly from making two separate (java) record classes with an interface for any properties common to both types.